crates/services/gas_price_service/src/v0/service.rs

```diff
@@ -194,17 +167,48 @@ mod tests {
         v0::{
             metadata::V0Metadata,
             service::GasPriceServiceV0,
-            uninitialized_task::initialize_algorithm,
+            uninitialized_task::{
+                initialize_algorithm,
+                SharedV0Algorithm,
+            },
         },
     };
     use fuel_core_services::{
+        RunnableService,
         Service,
         ServiceRunner,
+        StateWatcher,
     };
     use fuel_core_types::fuel_types::BlockHeight;
     use std::sync::Arc;
     use tokio::sync::mpsc;
 
+    #[async_trait::async_trait]
+    impl<L2, Metadata> RunnableService for GasPriceServiceV0<L2, Metadata>
```

Comment:
I don't see the point of testing code that is not used anywhere

Reply:
we're not testing the `RunnableService` it's just a helper for testing the logic within the task.

Reply:
I still don't see why do you need to implement `RunnableService`. The type already implements `RunnableTask`, and you need to test this one=)

Reply:
ref https://github.com/FuelLabs/fuel-core/pull/2317#discussion_r1792035909